test/units/testsuite-13.sh

```diff
@@ -53,6 +53,30 @@ function check_norbind {
     systemd-nspawn --register=no -D "$_root" --bind=/tmp/binddir:/mnt:norbind /bin/sh -c 'CONTENT=$(cat /mnt/subdir/file); if [[ $CONTENT != "outer" ]]; then echo "*** unexpected content: $CONTENT"; return 1; fi'
 }
 
+function check_rootidmap {
+    local _owner=1000
+    local _root="/var/lib/machines/testsuite-13.rootidmap-path"
+    rm -rf "$_root"
+
+    # Create ext4 image, as ext4 supports idmapped-mounts.
+    dd if=/dev/zero of=/tmp/ext4.img bs=4k count=2048
+    mkfs.ext4 /tmp/ext4.img
+    mkdir -p /tmp/rootidmapdir
+    mount /tmp/ext4.img /tmp/rootidmapdir
+
+    touch /tmp/rootidmapdir/file
+    chown -R $_owner:$_owner /tmp/rootidmapdir
+
+    /usr/lib/systemd/tests/testdata/create-busybox-container "$_root"
+    systemd-nspawn --register=no -D "$_root" --bind=/tmp/rootidmapdir:/mnt:rootidmap /bin/sh -c 'PERMISSIONS=$(stat -c "%u:%g" /mnt/file); if [[ $PERMISSIONS != "0:0" ]]; then echo "*** wrong permissions: $PERMISSIONS"; return 1; fi; touch /mnt/other_file'
```

Comment:
IIRC, CentOS8 does not support idmap mount. Please check the functionality is supported by the kernel before testing.

cc @mrc0mmand Please follow up if I commented something wrong.

Reply:
What about something like this:

```diff
diff --git a/test/units/testsuite-13.sh b/test/units/testsuite-13.sh
index 77c148158f..4ad7431e42 100755
--- a/test/units/testsuite-13.sh
+++ b/test/units/testsuite-13.sh
@@ -56,6 +56,7 @@ function check_norbind {
 function check_rootidmap {
     local _owner=1000
     local _root="/var/lib/machines/testsuite-13.rootidmap-path"
+    local _command
     rm -rf "$_root"
 
     # Create ext4 image, as ext4 supports idmapped-mounts.
@@ -68,7 +69,18 @@ function check_rootidmap {
     chown -R $_owner:$_owner /tmp/rootidmapdir
 
     /usr/lib/systemd/tests/testdata/create-busybox-container "$_root"
-    systemd-nspawn --register=no -D "$_root" --bind=/tmp/rootidmapdir:/mnt:rootidmap /bin/sh -c 'PERMISSIONS=$(stat -c "%u:%g" /mnt/file); if [[ $PERMISSIONS != "0:0" ]]; then echo "*** wrong permissions: $PERMISSIONS"; return 1; fi; touch /mnt/other_file'
+    _command='PERMISSIONS=$(stat -c "%u:%g" /mnt/file); if [[ $PERMISSIONS != "0:0" ]]; then echo "*** wrong permissions: $PERMISSIONS"; return 1; fi; touch /mnt/other_file'
+    if ! SYSTEMD_LOG_TARGET=console systemd-nspawn \
+            --register=no -D "$_root" \
+            --bind=/tmp/rootidmapdir:/mnt:rootidmap \
+            /bin/sh -c "$_command" |& tee nspawn.out; then
+        if grep -q "Failed to map ids for bind mount.*: Function not implemented" nspawn.out; then
+            echo "idmapped mounts are not supported, skipping the test..."
+            return 0
+        fi
+
+        return 1
+    fi
 
     PERMISSIONS=$(stat -c "%u:%g" /tmp/rootidmapdir/other_file)
     if [[ $PERMISSIONS != "$_owner:$_owner" ]]; then
```

I tested it both on C8S (without idmapped mount support) and F36 (with) and it seems to do the right thing, albeit it's not particularly pretty. I'm not sure if there's a way to check if idmapped mounts are supported without calling `mount_setattr()` directly.

Reply:
Applied @mrc0mmand's fix, thanks!